src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -233,6 +238,71 @@ public void markDirty()
         colony.markCitizensDirty();
     }
 
+    /**
+     * Create a CitizenData View given it's saved NBTTagCompound.
+     *
+     * @param id  The citizen's id.
+     * @param buf The network data.
+     * @return View object of the citizen.
+     */
+    @Nullable
+    public static CitizenDataView createCitizenDataView(final int id, final ByteBuf buf)
+    {
+        @Nullable CitizenDataView citizenDataView = new CitizenDataView(id);
+
+        try
+        {
+            citizenDataView.deserialize(buf);
+        }
+        catch (final RuntimeException ex)
+        {
+            Log.getLogger().error(String.format("A CitizenData.View for #%d has thrown an exception during loading, its state cannot be restored. Report this to the mod author",
+              citizenDataView.getId()), ex);
+            citizenDataView = null;
+        }
+
+        return citizenDataView;
+    }
+
+    /**
+     * Returns a random element in a list.
+     *
+     * @param rand  Random object.
+     * @param array Array to select from.
+     * @return Random element from array.
+     */
+    private static String getRandomElement(@NotNull final Random rand, @NotNull final String[] array)
+    {
+        return array[rand.nextInt(array.length)];
+    }
+
+    /**
+     * Returns a random capital letter from the alphabet.
+     *
+     * @param rand Random object.
+     * @return Random capital letter.
+     */
+    private static char getRandomLetter(@NotNull final Random rand)
+    {
+        return (char) (rand.nextInt(LETTERS_IN_THE_ALPHABET) + 'A');
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int result = id;
+        result = 31 * result + (name != null ? name.hashCode() : 0);
+        result = 31 * result + (female ? 1 : 0);
+        result = 31 * result + (colony != null ? colony.hashCode() : 0);
+        result = 31 * result + strength;
+        result = 31 * result + endurance;
+        result = 31 * result + charisma;
+        result = 31 * result + intelligence;
+        result = 31 * result + dexterity;
+        return result;
+    }
+
+    @SuppressWarnings(Suppression.TOO_MANY_RETURNS)
     @Override
     public boolean equals(final Object o)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)